client/my-sites/controller.js

```diff
@@ -169,7 +170,9 @@ module.exports = {
 		const onSelectedSiteAvailable = () => {
 			const selectedSite = sites.getSelectedSite();
 			siteStatsStickyTabActions.saveFilterAndSlug( false, selectedSite.slug );
-			context.store.dispatch( receiveSite( selectedSite ) );
+			if ( ! getSite( context.store.getState(), selectedSite.ID ) ) {
+				context.store.dispatch( receiveSite( selectedSite ) );
```

Comment:
This is an odd conditional. Would we be getting a `selectedSite` if it didn't already exist in the state tree?